makefiles/libc/newlib.mk

```diff
@@ -13,61 +14,61 @@ ifneq (,$(filter newlib_gnu_source,$(USEMODULE)))
-# Gentoo crossdev, but we prefer to look at /etc/alternatives first.
-# On OSX, newlib includes are possibly located in
-# /usr/local/opt/arm-none-eabi*/arm-none-eabi/include or /usr/local/opt/gcc-arm/arm-none-eabi/include
-NEWLIB_INCLUDE_PATTERNS ?= \
-  /etc/alternatives/gcc-$(TARGET_ARCH)-include \
-  /usr/$(TARGET_ARCH)/include \
-  /usr/local/opt/$(TARGET_ARCH)*/$(TARGET_ARCH)/include \
-  /usr/local/opt/gcc-*/$(TARGET_ARCH)/include \
-  #
-# Use the wildcard Makefile function to search for existing directories matching
-# the patterns above. We use the -isystem gcc/clang argument to add the include
-# directories as system include directories, which means they will not be
-# searched until after all the project specific include directories (-I/path)
-NEWLIB_INCLUDE_DIR ?= $(firstword $(wildcard $(NEWLIB_INCLUDE_PATTERNS)))
+  # Since Clang is not installed as a separate instance for each crossdev target
+  # we need to tell it where to look for platform specific includes (Newlib
+  # headers instead of Linux/Glibc headers.)
+  # On GCC this is done when building the cross compiler toolchain so we do not
+  # actually need to specify the include paths for system includes.
+  # Ubuntu gcc-arm-embedded toolchain (https://launchpad.net/gcc-arm-embedded)
+  # places newlib headers in several places, but the primary source seem to be
+  # /etc/alternatives/gcc-arm-none-eabi-include
+  # Gentoo Linux crossdev place the newlib headers in /usr/arm-none-eabi/include
+  # Arch Linux also place the newlib headers in /usr/arm-none-eabi/include
+  # Ubuntu seem to put a copy of the newlib headers in the same place as
+  # Gentoo crossdev, but we prefer to look at /etc/alternatives first.
+  # On OSX, newlib includes are possibly located in
+  # /usr/local/opt/arm-none-eabi*/arm-none-eabi/include or /usr/local/opt/gcc-arm/arm-none-eabi/include
+  NEWLIB_INCLUDE_PATTERNS ?= \
+    /etc/alternatives/gcc-$(TARGET_ARCH)-include \
+    /usr/$(TARGET_ARCH)/include \
+    /usr/local/opt/$(TARGET_ARCH)*/$(TARGET_ARCH)/include \
+    /usr/local/opt/gcc-*/$(TARGET_ARCH)/include \
+    #
 
-# If nothing was found we will try to fall back to searching for a cross-gcc in
-# the current PATH and use a relative path for the includes
-ifeq (,$(NEWLIB_INCLUDE_DIR))
-  NEWLIB_INCLUDE_DIR := $(abspath $(wildcard $(dir $(shell command -v $(PREFIX)gcc 2>/dev/null))/../$(TARGET_ARCH)/include))
-endif
+  # Search for a newlib.h inside the above newlib include directory patterns
+  NEWLIB_INCLUDE_FILE = $(firstword $(wildcard $(addsuffix /newlib.h,$(NEWLIB_INCLUDE_PATTERNS))))
 
-ifeq ($(TOOLCHAIN),llvm)
-  # A cross GCC already knows the target libc include path (build-in at compile time)
-  # but Clang, when cross-compiling, needs to be told where to find the headers
-  # for the system being built.
-  # We also add -nostdinc to avoid including the host system headers by mistake
-  # in case some header is missing from the cross tool chain
```

Comment:
This `ifeq` section with the `gcc` case was removed and it adds a wrong `-isystem` directory for mips-gcc 4.9.2 on my computer.

It supplies `/opt/mips-mti-elf/2016.05-03/mips-mti-elf/include/` which replaces `/opt/mips-mti-elf/2016.05-03/lib/gcc/mips-mti-elf/4.9.2/include/`

Which breaks compilation for mips in the `rmutex` example.

However, re-adding the `ifeq` would only solve my case but maybe not `llvm` as it would be using the wrong directory.

Reply:
Can you try if it works with the latest additions?